src/script/lua_api/l_server.cpp

```diff
@@ -479,26 +480,44 @@ int ModApiServer::l_dynamic_add_media(lua_State *L)
 		throw LuaError("Dynamic media cannot be added before server has started up");
 	Server *server = getServer(L);
 
+	bool fromdata = false;
+	bool ephemeral = false;
+
 	std::string filepath;
 	std::string to_player;
-	bool ephemeral = false;
+
+	std::string filename;
+	std::string filedata;
 
 	if (lua_istable(L, 1)) {
 		getstringfield(L, 1, "filepath", filepath);
 		getstringfield(L, 1, "to_player", to_player);
 		getboolfield(L, 1, "ephemeral", ephemeral);
+		getstringfield(L, 1, "filename", filename);
+		fromdata = getstringfield(L, 1, "data", filedata);
```

Comment:
Instead of copying here, you *could* use a `const char *` to the lua string, and the lua string's size, and pass that around.
Then you only have to memcpy the filedata once: When you put the string into the network packet.

In such cases I wish we had string views...

Reply:
`std::string_view` is C++17 :(
Is changing everything to a `const char *` (+keeping track of length) really the only way?
Would require re-working `addMediaFile`, as that expects a `&std::string` which it can assign to.